pkgs/by-name/op/openjfx/package.nix

```diff
@@ -69,25 +69,17 @@ stdenv.mkDerivation {
 
   inherit (source) src;
 
-  patches =
-    if featureVersion == "23" then
-      [
-        # 8338701: Provide media support for libavcodec version 61
-        # <https://github.com/openjdk/jfx23u/pull/18>
-        (fetchpatch2 {
-          url = "https://github.com/openjdk/jfx23u/commit/aba60fda1c82f00e8e685107592305c403a31287.patch?full_index=1";
-          hash = "sha256-+aRhTwi4VQthAq1SH1jxPl0mTosNMKoTY52jm+jiKso=";
-        })
-      ]
-    else if atLeast21 then
+  patches = lib.optionals (!atLeast23) (
+    if atLeast21 then
```

Comment:
Nitpick, but I think this double version check with an inversion makes the code kinda difficult to understand.

Reply:
Yeah, I’m still going back and forth on how to represent these multi‐versioned patches in these derivations. I have a prototype scheme that takes some inspiration from LLVM. The alternatives here would be `if featureVersion == "21" then … else if featureVersion == "17" then … else [ ]` or ` if atLeast21 && !atLeast23 then … else if atLeast17 then … else [ ]`, both of which are sort of awkward in their own ways. I’ll see if I can’t make it nicer for 25.05.